guides/svg/template.md

```diff
@@ -0,0 +1,274 @@
+            <text class="component-name" y="1em">{{ name }}</text>
+            <text class="component-type" y="2.5em">{{ definition.type }}</text>
+        </g>
+    </g>
+
+    <g transform="translate(0,50)">
+        <svg class="component-container" overflow="visible">
+            <rect class="container-background" width="100%" height="100%" fill="whitesmoke" />
+        </svg>
+    </g>
+    
+    {% if hasError %}
+    <g class="component-error">
+        <svg width="15" height="15" x="200" y="10" viewBox="0 0 100 100">
+          <circle fill="red" r="50" cx="50" cy="50"></circle>
+        </svg>
+    </g>
+    {% endif %}
+</svg>
+```
+
+### Why I see TWIG syntax in this model ?
+
+As you can see, we have some `twig` syntax in the template svg, **but** that's **not** `twig`. To render the template,
+we use the `nunjucks` templating engine. If you want to know more about it, you can read the
+[nunjucks documentation](https://mozilla.github.io/nunjucks/templating.html).
+
+With it, you can display all properties available in your
+[Component](https://ditrit.io/leto-modelizer-plugin-core/Component.html) instance. For example, you can display the
+component name with `{{ name }}` or the component type with `{{ definition.type }}`. 
+
+You can also use this functions :
+
+- `hasError` : return `true` if the component has an error, `false` otherwise.
+
+```jsx
+{% if hasError %}
+  <text class="component-error" y="2.5em">Error</text>
+{% endif %}
+```
+
+- `getAttribute(attributeName)` : return the attribute with the given name. For a component with the following attributes :
+
+```js
+const Component = {
+  name: 'MyComponent',
+  definition: {
+    type: 'MyType'
+  },
+  attributes: [
+    {
+      name: 'myAttributeName',
+      value: 'foo',
+    },
+    {
+      name: 'myOtherAttributeName',
+      value: 'bar',
+    },
+  ]
+}
+```
+
+You can display the value of the `myAttributeName` attribute like this :
+
+```jsx
+<text class="component-my-attribute-name" y="2.5em">{{ getAttribute('myAttributeName').value }}</text>
+```
+
+## Step by step
+
+### 1. Create the template :
+
+First you have to create the template with a `<svg></svg>` containing a class `template`. We also need to define the
+width and height for the drawer to work properly.
+
+```jsx
+<!-- Template -->
+<svg class="template" width="250" height="200"></svg>
+```
+
+### 2. Add the hitbox :
+
+The hitbox is the area where you can click to select or drag the component. It is usually a `<rect />` but it can be any
+shape you want.
+
+**Important :** you need to fill it with the `fill` attribute and set a `component-hitbox`class.
+
+```jsx
+<svg class="template" width="250" height="200">
+    <!-- Hitbox -->
+    <rect class="component-hitbox" width="100%" height="100%" fill="white" />
+</svg>
+```
+
+### 3. Add the icon container :
+
+The icon is the small picture that appears on the component. We define it by component definition type. It's usually a
```

Comment:
`We define it by component definition type.` -> not clear to me

Reply:
Done